bin/generate_protein_and_entity_ids.py

```diff
@@ -150,7 +150,7 @@ def main(args=None):
                         # (i.e. microbiome_bare_id) and assign the corresponding microbiome_id to the entities
                         for microbiome_id in microbiomes.groupby("microbiome_bare_id").get_group(microbiome_bare_id)[
                             "microbiome_id"
-                        ]:
+                        ].sort_values():
```

Comment:
Why was this still necessary? I thought the rows within one group should be anyway preserved, was this not the case?

Reply:
In this case, the `microbiomes.groupby("microbiome_bare_id")` is not deterministic as there will be multiples of the `microbiome_bare_id`, depending on if there are weights given for the same microbiome. Without the sorting here all tests containing multiple weights for the same microbiome e.g. `test_coassembly` will fail.